packages/demo-wallet-client/src/services/SmartWalletService.ts

```diff
@@ -81,10 +96,98 @@ export class SmartWalletService {
     }
   }
 
-  // private async generateFundedKeypair() {
-  //   const keypair = Keypair.random();
-  //   const server = new Server(SOROBAN_CONFIG.RPC_URL);
-  //   await server.requestAirdrop(keypair.publicKey());
-  //   return keypair;
-  // }
+  public async fundContractWithXLM(contractId: string) {
+    const fromAcc = await this.generateFundedKeypair();
+    return await this.transfer(
+      Asset.native().contractId(Networks.TESTNET),
+      fromAcc.publicKey(),
+      contractId,
+      9999,
+      fromAcc
+    );
+  }
+
+  public async transfer (
+    assetContract: string,
+    fromAcc: string,
+    toAcc: string,
+    amount: number,
+    signer: Keypair | SigningCallback,
+  ) {
+    const sourceAcc = await this.server.getAccount(this.sourceKeypair.publicKey());
+
+    // 1. Build transaction
+    const tx = new TransactionBuilder(
+      sourceAcc,
+      {
+        networkPassphrase: Networks.TESTNET,
+        fee: SOROBAN_CONFIG.MAX_FEE
+      })
+      .addOperation(Operation.invokeContractFunction({
+        contract: assetContract,
+        function: 'transfer',
+        args: [
+          nativeToScVal(fromAcc, { type: 'address' }), // from
+          nativeToScVal(toAcc, { type: 'address' }),  // to
+          nativeToScVal(amount * SOROBAN_CONFIG.ONE, { type: 'i128' }) // amount
+        ],
+      }))
+      .setTimeout(300)
+      .build();
+
+    // 2. Simulate transaction
+    const simulatedTx = await this.server.simulateTransaction(tx);
+    if (Api.isSimulationError(simulatedTx)) {
+      throw new Error(simulatedTx.error);
+    }
+
+    // 3. Sign auth entries
+    const signedEntries = await Promise.all(simulatedTx.result?.auth?.map(entry =>
+      authorizeEntry(entry, signer, simulatedTx.latestLedger + 600, Networks.TESTNET)
+    ) ?? []
+    );
+
+    // Step 4: Simulate again to get updated resources
```

Comment:
You shouldn't need to do this for a simple transfer :thinking: you should be able to just reassing `simulatedTx.result!.auth` to the `signedEntries` version and call `prepareTransaction`.